cores/arduino/stm32/analog.c

```diff
@@ -552,54 +653,86 @@ void HAL_ADC_MspDeInit(ADC_HandleTypeDef* hadc)
   */
 uint16_t adc_read_value(PinName pin)
 {
-  ADC_HandleTypeDef AdcHandle = {};
   ADC_ChannelConfTypeDef  AdcChannelConf = {};
   __IO uint16_t uhADCxConvertedValue = 0;
+  void *peripheral;
+  uint8_t indice = 0;
 
-  AdcHandle.Instance = pinmap_peripheral(pin, PinMap_ADC);
+  peripheral =  pinmap_peripheral(pin, PinMap_ADC);
 
-  if (AdcHandle.Instance == NP) return 0;
+  #ifdef ADC1_BASE
```

Comment:
getting indice from the peripheral is done in many places, couldn't you create a function where to get indice based on peripheral instance ? 

indice = get_indice(peripheral );
g_current_pinTab[indice] = pin;

that could be applied in many other places